### PR TITLE
MGDOBR-1295: refactoring error handling api

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-1295](https://issues.redhat.com/browse/MGDOBR-1295)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: infra-parent/infra-core/src/main/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/JsonMappingExceptionMapper.java [hunk 1]
```diff
@@ -25,14 +27,14 @@ protected JsonMappingExceptionMapper() {
...
-        super(bridgeErrorService, ExternalUserException.class);
+    public JsonMappingExceptionMapper(BridgeErrorService bridgeErrorService, Instance<ErrorHrefVersionProvider> builders) {
+        super(bridgeErrorService, ExternalUserException.class, builders);
     }
 
     @Override
     public Response toResponse(JsonMappingException e) {
-        LOGGER.debug("Failure", e);
         Throwable cause = Objects.nonNull(e.getCause()) ? e.getCause() : e;
+        // TODO: refactor, add tests
```
Comment (manstis) [inline infra-parent/infra-core/src/main/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/JsonMappingExceptionMapper.java:37]: Is this still relevant? You said on gChat that you'd done all tests.

File: infra-parent/infra-core/src/test/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/InternalPlatformExceptionMapperTest.java [hunk 2]
```diff
@@ -32,7 +32,8 @@ public class InternalPlatformExceptionMapperTest {
     @BeforeEach
     void setup() {
         when(bridgeErrorService.getError(InternalPlatformException.class)).thenReturn(Optional.of(MAPPED_ERROR));
-        this.mapper = new InternalPlatformExceptionMapper(bridgeErrorService);
+        // TODO: fix
```
Comment (manstis) [inline infra-parent/infra-core/src/test/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/InternalPlatformExceptionMapperTest.java:35]: What needs fixing?

File: manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/api/ManagerConstraintViolationExceptionMapper.java [hunk 3]
```diff
@@ -28,8 +29,6 @@ protected void init() {
 
     @Override
     protected ErrorResponse toErrorResponse(ConstraintViolation<?> cv) {
-        ErrorResponse errorResponse = super.toErrorResponse(cv);
-        errorResponse.setHref(V1APIConstants.V1_ERROR_API_BASE_PATH + errorResponse.getId());
-        return errorResponse;
+        return super.toErrorResponse(cv);
```
Comment (manstis) [inline manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/api/ManagerConstraintViolationExceptionMapper.java:32]: I guess you can remove the whole method?

File: manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/api/ManagerInternalPlatformExceptionMapper.java [hunk 4]
```diff
@@ -28,9 +29,7 @@ protected void init() {
 
     @Override
     protected ErrorResponse toErrorResponse(InternalPlatformException e) {
-        ErrorResponse errorResponse = super.toErrorResponse(e);
-        errorResponse.setHref(V1APIConstants.V1_ERROR_API_BASE_PATH + errorResponse.getId());
-        return errorResponse;
+        return super.toErrorResponse(e);
```
Comment (manstis) [inline manager-parent/manager-core/src/main/java/com/redhat/service/smartevents/manager/core/api/ManagerInternalPlatformExceptionMapper.java:32]: I guess you can remove this whole method too?

**Review by manstis — COMMENTED** [review]
Looks pretty good to me and looks exactly as I imagined. Great job.

A few minors; but nothing that should cause you any pain.

File: infra-parent/infra-core/src/main/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/JsonMappingExceptionMapper.java (hunk 1 above)
Comment (r00ta) [reply infra-parent/infra-core/src/main/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/JsonMappingExceptionMapper.java:37]: yup, I forgot to remove the comment :+1:

File: infra-parent/infra-core/src/test/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/InternalPlatformExceptionMapperTest.java (hunk 2 above)
Comment (r00ta) [reply infra-parent/infra-core/src/test/java/com/redhat/service/smartevents/infra/core/exceptions/mappers/InternalPlatformExceptionMapperTest.java:35]: Leftover :+1:

**Review by manstis — APPROVED** [review]
LGTM 👍

**Review by danielezonca — APPROVED** [review]
LGTM, well done :)

**Pull request merged**